src/sql/platform/telemetry/telemetryUtilities.ts

```diff
@@ -25,7 +24,8 @@ export function addTelemetry(
 	telemetryService: ITelemetryService,
 	telemetryEventName: string,
 	data?: IConnectionTelemetryData,
-	connection?: IConnectionProfile): Promise<void> {
+	connection?: IConnectionProfile,
+	logService?: ILogService): Promise<void> {
```

Comment:
I think this should be non-optional, right now you haven't updated any of the addTelemetry calls so they won't be logging anything at all. And there really isn't a good reason for the callers not to have a logging service.

And TBH I think this should really be getting the ITelemetryService itself and not requiring the caller to pass it in - none of the calls I saw did anything but use the injected VS service anyways so it seems kind of pointless to make each caller also have a dependency on that. But you don't have to consider that for this review if you don't want - if you would prefer to keep this scoped just let me know and I'll do that work separately.

Reply:
Yeah, I thought those other optionals were services at first glance which is why I wrote it this way.